design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,128 @@
+
+The following are mappings of functionalities for both Golang and Python CLIs.
+
+![golang-cli-mapping](../images/golang-cli-mapping.png)
+
+![python-cli-mapping](../images/python-cli-mapping.png)
+
+
+
+|            | Quality                                                 |
+| ---------- | ------------------------------------------------------- |
+| Ruby CLI   | Widely used<br />***Status***: Stable                   |
+| Go CLI     | Not "community" level certified <br />***Status***: POC |
+| Python CLI | ***Status***: Community/alpha level                     |
+
+|            | Deployment                                                   |
+| ---------- | ------------------------------------------------------------ |
+| Ruby CLI   | Options:<br />1. Deploy to a Docker container (not applicable for our use-case)<br />2. Deploy using a very specific version of Ruby<br />Which means that the CLI is not plug-and-play and has prerequisites of either Docker or Ruby. |
+| Go CLI     | The CLI is compiled down to a binary. A different binary is need to be compiled for each platform but no other requirements or installers are needed to launch the Go CLI. |
+| Python CLI | Compiled down to a single binary without the need for Python or other prerequisites to be installed on machine. |
+
+|            | Testing                                                      |
+| ---------- | ------------------------------------------------------------ |
+| Ruby CLI   | TBD                                                          |
+| Go CLI     | Pipeline maintained but the testing is not at the level it needs to be. Existing functionality will demand addition testing. Considering the limited implemented functionality, new tests will need to be added if we choose to expand. |
+| Python CLI | Explanation TBD but tested well                              |
+
+|            | Security |
+| ---------- | -------- |
+| Ruby CLI   | TBD      |
+| Go CLI     | TBD      |
+| Python CLI | TBD      |
+
+#### Estimation
+
+Ruby CLI - TBD
+
+Golang CLI - TBD
+
+Python CLI - TBD
+
+
+
+#### Takeways and recommendation
+
+Golang CLI would require a full rewrite due to it's limited functionality, test coverage, and architectural flaws. 
+
+Ruby requires the least amount of work but its UX is not intuitive and as part of its deployment, requires the installation of Docker (which is not valid for our use-case) or a specific version of Ruby to run the CLI. 
+
+Python has most of the basic actions needed to interface with the Conjur server. We will need to fill in the gaps but is architecturally more stable and well tested. The Python CLI is compiled down into a single binary so the user will not need install any additional dependencies.
+
+*Recommendation*: Python CLI because of its better design and it acts as a great point of entry.
```

Comment:
What is the criteria on which we are comparing between the options? Can you please specify them and rank the solutions based on them?

Also, I am missing the discussion about the effort size. What is the effort that each requires to bring it to GA quality and robustness?

Reply:
> What is the criteria on which we are comparing between the options?

We are comparing them in terms of functionalities, quality, ease of use (ux), deployment, testing, security.

> Can you please specify them and rank the solutions based on them?

Yes!

> Also, I am missing the discussion about the effort size. What is the effort that each requires to bring it to GA quality and robustness?

Will get back to you on that once I have researched a bit more but adding it to my open questions